src/Toolkit/Str.php

```diff
@@ -600,6 +600,31 @@ public static function kebabToCamel(string|null $value): string
 		));
 	}
 
+
+	public static function label(string $value): string
+	{
+		// replace punctuation with spaces
+		$value = str_replace(['_', '-', '.'], ' ', $value);
+
+		// add a space before every uppercase character
```

Comment:
`Suggestion`: I first had to wrap my head around this RegEx, so for future reference:

```suggestion
		// add a space before every uppercase character by matching
		// all characters that are not Unicode lowercase or numbers
```